app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/dataset_editing/DisplayDatasetFragment.kt

```diff
@@ -140,42 +180,34 @@ class DisplayDatasetActivity : AppCompatActivity() {
 
     }
 
-    private fun checkIntentExtras(extras: Bundle?) {
-        lifecycleScope.launch {
-            datasetId = if (extras != null && extras["dataset_id"] is String) {
-                intent.getStringExtra("dataset_id")!!
-            } else {
-                "uEwDkGoGADW4hEJoJ6BA"
-            }
-        }
-    }
-
     private fun setGridViewItemListener() {
-        findViewById<GridView>(R.id.display_dataset_imagesGridView).setOnItemClickListener { _, _, i, _ ->
-            val cat = categories.elementAt(i)
-            val intent =
-                Intent(this@DisplayDatasetActivity, DisplayImageSetActivity::class.java)
-            intent.putExtra(
-                "category_of_pictures",
-                cat
-            )
-            intent.putExtra("dataset_id", datasetId)
-            startActivity(intent)
-        }
+        parentActivity.findViewById<GridView>(R.id.display_dataset_imagesGridView)
+            .setOnItemClickListener { adapterView, view, i, l ->
+                val category = categories.elementAt(i)
+                val action = DisplayDatasetFragmentDirections.actionDisplayDatasetFragmentToDisplayImageSetFragment(datasetId, category)
+                findNavController().navigate(action)
+            }
     }
 
     private fun setTextChangeListener() {
-        findViewById<EditText>(R.id.display_dataset_name).doAfterTextChanged {
+        parentActivity.findViewById<EditText>(R.id.display_dataset_name).doAfterTextChanged {
             lifecycleScope.launch {
                 dataset = dbManagement.editDatasetName(
                     dataset,
-                    findViewById<EditText>(R.id.display_dataset_name).text.toString()
+                    binding.displayDatasetName.text.toString()
                 )
             }
         }
     }
 
-    override fun onBackPressed() {
-        startActivity(Intent(this, DatasetsOverviewActivity::class.java))
+    private fun <T>Fragment.getNavigationResult(key: String) =
```

Comment:
What's the reasoning behind this and why is it an extension method?

Reply:
That's what I used to retrieve data from addPicture initially, but it's not working so now I use regular navigation. I'll delete this.